src/app/gateways/views/DelegatedGatewayDetailView.vue

```diff
@@ -77,17 +67,14 @@
               <h3>{{ t('delegated-gateways.detail.data_plane_proxies') }}</h3>
 
               <KCard class="mt-4">
-                <DataSource
+                <DataLoader
                   v-slot="{ data: dataplanesData, error: dataplanesError }: DataplaneOverviewCollectionSource"
                   :src="`/meshes/${route.params.mesh}/dataplanes/for/${route.params.service}?page=${route.params.page}&size=${route.params.size}&search=${route.params.s}`"
+                  :data="[data]"
+                  :errors="[error]"
                 >
-                  <ErrorBlock
-                    v-if="dataplanesError !== undefined"
-                    :error="dataplanesError"
-                  />
-
                   <AppCollection
-                    v-else
+                    v-if="dataplanesData"
```

Comment:
Ah this is the thing we have to be a little careful of. AppCollection has its own loader/skeleton, which is very coupled to its look/feel. So we need to use AppCollections loader but use DataLoaders errors, so here you should be able to use `DataLoader::loader="false"` to render DataLoaders default slot even when loading ..... aaaaaah ok, right this is a reason why we should use DataSource for the other conversation I just resolved 😅 

Yeah I just tried, so in both places where we load a AppCollection in this PR, use `DataLoader::loader="false"` and remove the `v-if="dataplanesData/data"`. In the above conversation you'll also need to implement the suggestion I had to use DataSource at the top-level.

Once you've done this you'll see we have a skeleton loader from AppCollection instead of a spinner.

Sorry for the confusion, lmk if this isn't clear.

Reply:
Hm, this is a pretty inconvenient state when interacting with DataLoader. DataSource doesn’t include an error block so with that it’s okay to reason about whether you need to put an error block in the view component or let AppCollection handle it via KTable. But with DataLoader, one should be able to make the assumption that one _definitely_ doesn’t have to deal with that question. But I guess that’s owed to the fact that we can’t just stop using AppCollection’s (read: KTable’s) loading state stuff (though I’d rather use ours).

I tried implementing these suggestions in the last suggestion.

Reply:
Yeah this:

> But I guess that’s owed to the fact that we can’t just stop using AppCollection’s (read: KTable’s) loading state stuff

--

Its a little awkward, but the loading state is tied to AppCollection because the skeleton needs to look like a table. I thin it might look more awkward here because the loading logic is a tiny bit more involved. If you take a look at `MeshListView` its way less awkward/more straightforwards. Its just the addition of a `loader="false"`.

Let me take another look at this with it switched up.

Reply:
K I think this looks good now, let me take one last scan and I think I can 👍 . 2tics

Reply:
Ok, it looks like BuiltIn if fine but delegated I think needs a tiny bit more work. If you set `KUMA_LATENCY=3000` you'll see this:

![Screenshot 2024-03-01 at 13 11 58](https://github.com/kumahq/kuma-gui/assets/554604/b822d47d-bdb9-42a9-9e8d-498b1b6a6db5)

At a first glance, you _may_ need 2 DataLoaders here, a spinner for the top content and a skeleton via AppCollection for the list of dataplanes underneath.

I might be mistaken though, let me know either way

Reply:
What I did here now was to move the outer data source for the ServiceInsight further in so that it only wraps around the top card. I changed it to a DataLoader because it needs to have its independent loading and error handling.

I also changed the AppCollection still having a `v-if` which prevented the KTable loading mechanism to handle the loading state. It’s also a DataLoader but one with `:loader="false"`.

Reply:
gotcha lemme take another look in a sec